tasks/collect-slack-notification-params/collect-slack-notification-params.yaml

```diff
@@ -0,0 +1,180 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: collect-slack-notification-params
+  labels:
+    app.kubernetes.io/version: "0.0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task that collects slack notification configuration options from Release CRs and the data file
+  params:
+    - name: release
+      type: string
+      description:
+        The namespaced name (namespace/name) of the Release custom resource initiating this pipeline execution
+    - name: releaseplanadmission
+      type: string
+      description: The namespaced name (namespace/name) of the releasePlanAdmission
+    - name: releasepipelinename
+      description: name of the release pipeline
+      type: string
+    - name: snapshotPath
+      description: Path to the JSON string of the mapped Snapshot spec in the data workspace
+      type: string
+      default: "mapped_snapshot.json"
+    - name: dataPath
+      type: string
+      description: Path to the JSON string of the merged data containing the configuration options to use
+      default: data.json
+    - name: hacUrl
+      type: string
+      description: Root URL of HAC UI (hybrid access console)
+      default: "https://console.redhat.com/preview/application-pipeline/workspaces"
+  workspaces:
+    - name: data
+      description: The workspace where the snapshot spec json file resides
+  results:
+    - name: message
+      type: string
+      description: the formatted message to use in downstream tasks
+  steps:
+    - name: create-message
+      image: quay.io/hacbs-release/release-utils:85ab98a7ec63c3d8d9ec3c4982ff0e581bcb3c83
+      script: |
+        #!/usr/bin/env sh
+        set -x
+
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No valid data file was provided."
+            exit 1
+        fi
+        SNAPSHOT_FILE="$(workspaces.data.path)/$(params.snapshotPath)"
+        if [ ! -f "${SNAPSHOT_FILE}" ] ; then
+            echo "No valid snapshot file was provided."
+            exit 1
+        fi
+
+        if [ $(jq '.slack | has("channelId")' $DATA_FILE) == false ] ; then
+            echo "Slack channel ID missing in data JSON file - returning empty message"
+            echo "" > $(results.message.path)
+            exit
+        fi
+
+        channel_id=$(jq -jr '.slack.channelId' $DATA_FILE)
+        
+        origin_namespace=$(echo $(params.release) | cut -f1 -d/)
+        target_namespace=$(echo $(params.releaseplanadmission) | cut -f1 -d/)
+        
+        origin_workspace=$(echo $origin_namespace | sed 's/_tenant//g')   
+        target_workspace=$(echo $target_namespace | sed 's/_tenant//g')
+        
+        release_name=$(echo $(params.release) | cut -f2 -d/)
+        release_pipeline_name="$(echo $(params.releasepipelinename))"
```

Comment:
then we need to change the image we use for this task for the one what has `oc` or `kubectl` in it...which does not have `jq` in it. So this task would need to be split into more steps. Seems more complicated.

Reply:
It may be more complicated, but we don't need an image switch. We just need the `release.json` like we have the snapshot file
`origin_namespace=$(jq -jr '.metadata.namespace' $RELEASE_FILE)`
`release_name=$(jq -jr '.metadata.name' $RELEASE_FILE)`
`release_pipeline_name="$(jq -jr '.status.IForgetThePathHere` $RELEASE_FILE)"`
@mmalina would have to double check me if the `-j` is needed